Marshal/Marshaling.swift

```diff
@@ -14,5 +14,5 @@
 import Foundation
 
 public protocol Marshaling {
-    func marshal() -> MarshaledObject
+    func marshal() -> Marshal
 }
```

Comment:
Does the `Marshaling` protocol make sense now that the core is a protocol? Every time I've implemented and used `Marshaling` it was to convert Object -> Dictionary -> NSData (JSON encoded), not to create a dictionary suitable to pass to another `init(object:)`. 

That means that the implementation of `marshal()` needs to know what type of object to return, and cannot actually return any type conforming to `Marshal` (or whatever is the ultimate name of the core protocol).